packages/extension-ui/src/Popup/Contacts/AddContact.tsx

```diff
@@ -0,0 +1,376 @@
+import { Contact, Identity } from '@polkadot/extension-base/background/types';
+import { ContactsStore } from '@polkadot/extension-base/stores';
+import { decodeAddress, encodeAddress } from '@polkadot/keyring';
+import { hexToString, hexToU8a, isHex } from '@polkadot/util';
+
+import { ActionBar, ActionContext, ActionText, Button, InputWithLabel } from '../../components';
+import useTranslation from '../../hooks/useTranslation';
+import { Header } from '../../partials';
+import chains from '../../util/chains';
+
+const bs58 = require('bs58');
+
+/**
+ * Get address prefix
+ * Polkadot: 0
+ * Kusama: 2
+ * Substrate: 42
+ * others: ...
+ * @param address
+ */
+function getAddressPrefix (address: string): number {
+  const bytes = bs58.decode(address);
+  const hex = bytes.toString('hex');
+  const prefix = `${hex[0]}${hex[1]}`;
+
+  return parseInt(prefix, 16);
+}
+
+/**
+ * Check address is valid address
+ * @param address
+ */
+function isValidAddressPolkadotAddress (address: string): boolean {
+  try {
+    encodeAddress(
+      isHex(address)
+        ? hexToU8a(address)
+        : decodeAddress(address)
+    );
+
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+/**
+ * format the identity response to identity object: { isBad, isGood, info... }
+ * @param identity
+ */
+function formatIdentity (identity: Record<string, string>): Identity {
+  if (_.isEmpty(identity)) {
+    return {
+      isBad: true,
+      isGood: false,
+      info: {}
+    } as Identity;
+  }
+
+  const { info, judgements } = identity;
+  const isKnownGood = judgements.some(([, judgement]) => Object.prototype.hasOwnProperty.call(judgement, 'Known Good'));
+  const isReasonable = judgements.some(([, judgement]) => Object.prototype.hasOwnProperty.call(judgement, 'Reasonable'));
+  const isLowQuality = judgements.some(([, judgement]) => Object.prototype.hasOwnProperty.call(judgement, 'Low Qualit'));
+  const isErroneous = judgements.some(([, judgement]) => Object.prototype.hasOwnProperty.call(judgement, 'Erroneous'));
+
+  return {
+    isBad: isLowQuality || isErroneous,
+    isGood: isKnownGood || isReasonable,
+    info: {
+      Display: info.display.Raw ? hexToString(info.display.Raw) : '',
+      Legal: info.legal.Raw ? hexToString(info.legal.Raw) : '',
+      Email: info.email.Raw ? hexToString(info.email.Raw) : '',
+      Web: info.web.Raw ? hexToString(info.web.Raw) : '',
+      Twitter: info.twitter.Raw ? hexToString(info.twitter.Raw) : '',
+      Riot: info.riot.Raw ? hexToString(info.riot.Raw) : ''
+    }
+  } as Identity;
+}
+
+interface Chain {
+  chain: string;
+  genesisHash?: string;
+  icon: string;
+  ss58Format: number;
+}
+
+interface Props extends ThemeProps {
+  className?: string;
+}
+
+/**
+ *  The chain's endpoint list.
+ *  [prefix]: [endpoint]
+ *  endpoint is '' means that I did not find the endpoint in the Polkadot-js webapp's config file.
+ *  */
+const ChainsEndPoint = {
```

Comment:
No, the extension cannot have any network connectivity at all. It breaks the model completely. Additionally all of these chains have specific types - we just don't provide them and this will break.

Reply:
Got it, we can keep these off of the extension, so it's the user's responsibility to choose a trusted website and then to interact with network, right? I'm thinking whether is there any guidance/whitelisting that the extension can do while isolating the liability.